fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -381,8 +420,17 @@ private OptExpression logicalRuleRewrite(
         SessionVariable sessionVariable = rootTaskContext.getOptimizerContext().getSessionVariable();
         CTEContext cteContext = context.getCteContext();
         CTEUtils.collectCteOperators(tree, context);
-        if (sessionVariable.isEnableRboTablePrune()) {
-            context.setEnableLeftRightJoinEquivalenceDerive(false);
+
+        // Whether to do complete equivalence derive in
+        // {@link com.starrocks.sql.optimizer.rule.transformation.PushDownJoinOnClauseRule}, eg: outer join to inner join,
+        // or derive equivalence derive for outer joins.
+        // NOTE: It's useful for normal queries but it will disturb some rewrite rule(eg: mv rewrite, table prune), so add
+        // a config to control it.
+        if (sessionVariable.isEnableRboTablePrune() || mvRewriteStrategy.mvStrategy.isMultiStages()) {
+            context.getJoinPushDownParams().enableLeftRightJoinEquivalenceDerive = false;
```

Comment:
what if the plan is not rewritten by mv, then we can not convert left join to inner join?

Reply:
no.
After `        context.getJoinPushDownParams().reset();`,         ` ruleRewriteIterative(tree, rootTaskContext, RuleSetType.PUSH_DOWN_PREDICATE);` still can trigger outer join push down in the following.